# 2025 June 25: Copied from OpenBLAS upstream commit:
# https://github.com/OpenMathLib/OpenBLAS/commit/fdc1c323404be6f9551b10ea11b6c4cd7254e9b0
# https://github.com/OpenMathLib/OpenBLAS/pull/5336
#
# Supposed to fix "Argument list too long" when linking dylib
# on earlier macOS versions.
#
# Applies to OpenBLAS 0.3.30, also maybe earlier versions.
# Remove at next upstream release, should be fixed then.

--- orig/CMakeLists.txt	2025-06-19 03:45:39
+++ CMakeLists.txt	2025-06-25 15:46:54
@@ -305,8 +305,8 @@
   endif()
 endif()
 
-# Fix "Argument list too long" for macOS with Intel CPUs and DYNAMIC_ARCH turned on
-if(APPLE AND DYNAMIC_ARCH AND (NOT CMAKE_HOST_SYSTEM_PROCESSOR STREQUAL "arm64"))
+# Fix "Argument list too long" for macOS with POWERPC or Intel CPUs 
+if(APPLE AND (NOT CMAKE_HOST_SYSTEM_PROCESSOR STREQUAL "arm64"))
   # Use response files
   set(CMAKE_C_USE_RESPONSE_FILE_FOR_OBJECTS 1)
   # Always build static library first
